homeassistant/components/cover/group.py

```diff
@@ -0,0 +1,271 @@
+    async_add_devices(
+        [CoverGroup(config[CONF_NAME], config[CONF_ENTITIES])])
+
+
+class CoverGroup(CoverDevice):
+    """Representation of a CoverGroup."""
+
+    def __init__(self, name, entities):
+        """Initialize a CoverGroup entity."""
+        self._name = name
+        self._tilt = False
+
+        self._is_closed = False
+        self._cover_position = 100
+        self._tilt_position = None
+        self._supported_features = 0
+
+        self._entities = entities
+        self._covers = {KEY_OPEN_CLOSE: set(), KEY_STOP: set(),
+                        KEY_POSITION: set()}
+        self._tilts = {KEY_OPEN_CLOSE: set(), KEY_STOP: set(),
+                       KEY_POSITION: set()}
+
+    def update_supported_features(self, entity_id, old_state, new_state):
+        """Update dictionaries with supported features."""
+        if new_state is None:
+            for value in self._covers.values():
+                value.discard(entity_id)
+            for value in self._tilts.values():
+                value.discard(entity_id)
+                tilt = True if value else False
+            self._tilt = tilt
+            return
+
+        if old_state is None:
+            features = new_state.attributes[ATTR_SUPPORTED_FEATURES]
+            if features & (SUPPORT_OPEN | SUPPORT_CLOSE):
+                self._covers[KEY_OPEN_CLOSE].add(entity_id)
+            if features & (SUPPORT_STOP):
+                self._covers[KEY_STOP].add(entity_id)
+            if features & (SUPPORT_SET_POSITION):
+                self._covers[KEY_POSITION].add(entity_id)
+            if features & (SUPPORT_OPEN_TILT | SUPPORT_CLOSE_TILT):
+                self._tilts[KEY_OPEN_CLOSE].add(entity_id)
+                self._tilt = True
+            if features & (SUPPORT_STOP_TILT):
+                self._tilts[KEY_STOP].add(entity_id)
+                self._tilt = True
+            if features & (SUPPORT_SET_TILT_POSITION):
+                self._tilts[KEY_POSITION].add(entity_id)
+                self._tilt = True
+
+    async def async_added_to_hass(self):
+        """Register callbacks."""
+        @callback
+        def state_change_listener(entity, old_state, new_state):
+            """Handle cover state changes and update state."""
+            self.update_supported_features(entity, old_state, new_state)
+            self.async_schedule_update_ha_state(True)
+
+        for entity_id in self._entities:
+            new_state = self.hass.states.get(entity_id)
+            self.update_supported_features(entity_id, None, new_state)
+        self.async_schedule_update_ha_state(True)
+        async_track_state_change(self.hass, self._entities,
+                                 state_change_listener)
+
+    @property
+    def name(self):
+        """Return the name of the cover."""
+        return self._name
+
+    @property
+    def assumed_state(self):
+        """Enable buttons even if at end position."""
+        return True
```

Comment:
This is only an assumed state if any of the children is an assumed state. Otherwise we know all the states that this state depends on and thus it's not assumed.